core/variant_setget.cpp

```diff
@@ -33,1994 +33,1377 @@
+			const m_base_type &v = *reinterpret_cast<const m_base_type *>(base);                                                    \
+			const Variant *ptr = v.getptr(index);                                                                                   \
+			NULL_TEST(ptr);                                                                                                         \
+			PtrToArg<Variant>::encode(*ptr, member);                                                                                \
+		}                                                                                                                           \
+		static void set(Variant *base, int64_t index, const Variant *value, bool &valid, bool &oob) {                               \
+			(*VariantGetInternalPtr<m_base_type>::get_ptr(base))[index] = *value;                                                   \
+			oob = false;                                                                                                            \
+			valid = true;                                                                                                           \
+		}                                                                                                                           \
+		static void validated_set(Variant *base, int64_t index, const Variant *value, bool &oob) {                                  \
+			(*VariantGetInternalPtr<m_base_type>::get_ptr(base))[index] = *value;                                                   \
+			oob = false;                                                                                                            \
+		}                                                                                                                           \
+		static void ptr_set(void *base, int64_t index, const void *member) {                                                        \
+			m_base_type &v = *reinterpret_cast<m_base_type *>(base);                                                                \
+			v[index] = PtrToArg<Variant>::convert(member);                                                                          \
+		}                                                                                                                           \
+		static Variant::Type get_index_type() { return Variant::NIL; }                                                              \
+		static uint64_t get_indexed_size(const Variant *base) { return VariantGetInternalPtr<m_base_type>::get_ptr(base)->size(); } \
+	};
+
+INDEXED_SETGET_STRUCT_BULTIN_NUMERIC(Vector2, double, real_t, 2)
+INDEXED_SETGET_STRUCT_BULTIN_NUMERIC(Vector2i, int64_t, int32_t, 2)
+INDEXED_SETGET_STRUCT_BULTIN_NUMERIC(Vector3, double, real_t, 3)
+INDEXED_SETGET_STRUCT_BULTIN_NUMERIC(Vector3i, int64_t, int32_t, 3)
+INDEXED_SETGET_STRUCT_BULTIN_NUMERIC(Quat, double, real_t, 4)
+INDEXED_SETGET_STRUCT_BULTIN_NUMERIC(Color, double, real_t, 4)
```

Comment:
```suggestion
INDEXED_SETGET_STRUCT_BULTIN_NUMERIC(Color, double, float, 4)
```

Colors should be always explicitly single-precision, since there is no use case for them to have more than 23 significant digits.